Makefile

```diff
@@ -0,0 +1,101 @@
+GOCMD=go
+GOTEST=$(GOCMD) test
+GOVET=$(GOCMD) vet
+BINARY_NAME=Spinup
+VERSION?=0.0.0
+SERVICE_PORT?=3000
+DOCKER_REGISTRY?= #if set it should finished by /
+EXPORT_RESULT?=false # for CI please set EXPORT_RESULT to true
+
+GREEN  := $(shell tput -Txterm setaf 2)
+YELLOW := $(shell tput -Txterm setaf 3)
+WHITE  := $(shell tput -Txterm setaf 7)
+CYAN   := $(shell tput -Txterm setaf 6)
+RESET  := $(shell tput -Txterm sgr0)
+
+.PHONY: all test build
+
+all: help
+
+## Build:
+build: ## Build your project and put the output binary in out/bin/
+	mkdir -p out/bin
+	GO111MODULE=on $(GOCMD) build -o out/bin/$(BINARY_NAME) .
+
+clean: ## Remove build related file 
+	rm -fr ./bin
+	rm -fr ./out
+	rm -f ./junit-report.xml checkstyle-report.xml ./coverage.xml ./profile.cov yamllint-checkstyle.xml
+
+
+watch: ## Start the Spinup -. FULL in watch model
+	air
+
+watch-install: ## Install Watch Tools
+	go install github.com/cosmtrek/air@master
+## Test:
+test: ## Run the tests of the project
+ifeq ($(EXPORT_RESULT), true)
+	GO111MODULE=off go get -u github.com/jstemmer/go-junit-report
+	$(eval OUTPUT_OPTIONS = | tee /dev/tty | go-junit-report -set-exit-code > junit-report.xml)
+endif
+	$(GOTEST) -v -race ./... $(OUTPUT_OPTIONS)
+
+coverage: ## Run the tests of the project and export the coverage
+	$(GOTEST) -cover -covermode=count -coverprofile=profile.cov ./...
+	$(GOCMD) tool cover -func profile.cov
+ifeq ($(EXPORT_RESULT), true)
+	GO111MODULE=off go get -u github.com/AlekSi/gocov-xml
+	GO111MODULE=off go get -u github.com/axw/gocov/gocov
+	gocov convert profile.cov | gocov-xml > coverage.xml
+endif
+
+## Dependencies:
+dep:  ## download your dependencies
+	go mod download
+
+## Server :
+server-start:  ## Start your Server from the build output.  
+	./out/bin/Spinup start
+
+server-startapi:  ## Start your Server from the build output API ONLY. 
+	./out/bin/Spinup start --api-only
+
+## Run :
+run:  ## Start your Server from the curent  code.  
+	go run ./main.go start
+
+runapi:  ## Start your Server from the code API Only.  
+	go run ./main.go start --api-only
+
+## Lint:
+lint: lint-go lint-dockerfile ## Run all available linters
+
+lint-go-install: ## Install golang-ci lint
+	go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.45.2
+
+lint-go: ## Use golintci-lint on your project
+	golangci-lint run --enable-all
+
+## Docker:
+docker-build: ## Use the dockerfile to build the container
+	docker build --rm --tag $(BINARY_NAME) .
+
+docker-release: ## Release the container with tag latest and version
+	docker tag $(BINARY_NAME) $(DOCKER_REGISTRY)$(BINARY_NAME):latest
+	docker tag $(BINARY_NAME) $(DOCKER_REGISTRY)$(BINARY_NAME):$(VERSION)
+	# Push the docker images
+	docker push $(DOCKER_REGISTRY)$(BINARY_NAME):latest
+	docker push $(DOCKER_REGISTRY)$(BINARY_NAME):$(VERSION)
```

Comment:
Not sure, but docker support might not come up any time soon, as spinup depends on the host machine having docker installed. I think we can remove this for now.